lib/checks/aria/aria-allowed-attr-evaluate.js

```diff
@@ -27,31 +33,80 @@ import { getRole, allowedAttr, validateAttr } from '../../commons/aria';
  */
 function ariaAllowedAttrEvaluate(node, options, virtualNode) {
   const invalid = [];
-
+  const rowAttrInvalid = [];
   const role = getRole(virtualNode);
   const attrs = virtualNode.attrNames;
   let allowed = allowedAttr(role);
-
   // @deprecated: allowed attr options to pass more attrs.
   // configure the standards spec instead
   if (Array.isArray(options[role])) {
     allowed = uniqueArray(options[role].concat(allowed));
   }
 
+  let tableRole;
+  function validateRowAttrs(validValue) {
```

Comment:
I kinda feel like this should be hoisted out of this parent funtion.

Reply:
`tableRole` doesn't need to be outside of `validateRowAttrs` after doing some clean up. However, I think we can't hoist that block completely out of the function as we use that variable for setting the cache and `closest()` which we only do if the `virtualNode` has a parent and the role is a row.